precompiles/preimage/src/lib.rs

```diff
@@ -56,15 +57,16 @@ where
 	fn note_preimage(
 		handle: &mut impl PrecompileHandle,
 		encoded_proposal: BoundedBytes<GetEncodedProposalSizeLimit>,
-	) -> EvmResult {
-		let bytes = encoded_proposal.into();
+	) -> EvmResult<H256> {
+		let bytes: Vec<u8> = encoded_proposal.into();
+		let hash: H256 = Runtime::Hashing::hash(&bytes).into();
```

Comment:
It is not ideal to perform the hash twice but I suppose it is acceptable considering it is not a widely used call and would require a lot of rework in substrate to better support it